quiche/src/lib.rs

```diff
@@ -2071,6 +2081,27 @@ impl Connection {
         Ok(())
     }
 
+    /// Returns true if a QUIC packet is a stateless reset.
+    fn is_stateless_reset(&self, buf: &[u8]) -> bool {
+        // If the packet is too small, then we just throw it away
+        let buf_len = buf.len();
+        if buf_len < 21 {
+            return false;
+        }
+
+        match self.peer_transport_params.stateless_reset_token {
```

Comment:
In theory we should check against the reset token of every active connection ID here, not just the initial one. Though it doesn't look like we have an API for iterating over all connection IDs currently, so the current code is probably fine for now, but let's add a TODO comment here for future reference.
```suggestion
        // TODO: we should iterate over all active destination connection IDs and check against their reset token.
        match self.peer_transport_params.stateless_reset_token {
```

Reply:
Thanks for the explanation and the code link! I'll be watching out for these cases.